asyncpg/protocol/record/recordobj.c

```diff
@@ -51,7 +66,6 @@ ApgRecord_New(PyObject *desc, Py_ssize_t size)
     Py_INCREF(desc);
     o->desc = (ApgRecordDescObject*)desc;
     o->self_hash = -1;
-    PyObject_GC_Track(o);
```

Comment:
Hm, would an asyncpg.Record subclass implement gc track?

Reply:
Would a subclass inherit the Record's dealloc which messes with the freelist?

Reply:
> Hm, would an asyncpg.Record subclass implement gc track?

The default `object` `tp_alloc` does `GC_Track`.

> Would a subclass inherit the Record's dealloc which messes with the freelist?

There's an explicit check there (`ApgRecord_CheckExact`):
``
        if (len < ApgRecord_MAXSAVESIZE &&
            numfree[len] < ApgRecord_MAXFREELIST &&
            ApgRecord_CheckExact(o))
```

Reply:
Can you add an assert calling _PyObject_GC_IS_TRACKED?

Reply:
Better to check if `PyObject_GC_IS_TRACKED` is defined (it is in 3.9) and use it, if not use `_PyObject_GC_IS_TRACKED` instead. I'd like to have this assertion as if somehow GCTrack call is missing it's a world of pain to debug that.

Reply:
> Better to check if `PyObject_GC_IS_TRACKED` is defined (it is in 3.9) 

I only see the underscore version in 3.9 (and master).  Are you sure it's defined?

Reply:
Apparently not. Let's use the underscore version if available. 🤷‍♂️